### PR TITLE
Fix tile dumping when /tmp is not temporary directory

### PR DESCRIPTION
**Pull request by Minion3665**
On iOS, each app has its own sandbox which has a temporary directory. The FileUtil::getSysTempDirectoryPath() method successfully gets this directory and we create a tiledump directory under it. Sadly, we then try to write to /tmp regardless of where the temporary directory is.

This commit makes us also write our file in the directory returned by FileUtil::getSysTempDirectoryPath()

Change-Id: Ic8425cb94d1d85ac9c77212eb84d0ca7c46cd34e

* Target version: master 

### TODO

- [ ] ...

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Review by mmeeks — APPROVED** [review]
This is only enabled in a compile-time debugging mode, and it's called only after dropping privileges from the Kit process - even in this case, so I'm not concerned.

File: kit/Delta.hpp [hunk 1]
```diff
@@ -734,45 +734,40 @@
...
                 }
-
-                std::ofstream tileFile(path, std::ios::binary);
-                std::vector<char> pngOutput;
-                Png::encodeSubBufferToPNG(pixmap, startX, startY, width, height,
-                                        bufferWidth, bufferHeight, pngOutput, mode);
-                tileFile.write(pngOutput.data(), pngOutput.size());
             }
+
+            std::ofstream tileFile(path, std::ios::binary);
```
[the remark this answers is not in the document]
Comment (mmeeks) [reply kit/Delta.hpp:766]: As per my comment - it looks fine; thanks for asking Skyler.

**Review by OFFTKP — CHANGES_REQUESTED** [review]

File: kit/Delta.hpp [hunk 2]
```diff
@@ -734,45 +734,40 @@ class DeltaGenerator {
         int dumpedIndex = 1;
         if (dumpTiles)
         {
-            std::string path = "/tmp/tiledump";
-            bool directoryExists = !FileUtil::isEmptyDirectory(path);
+            std::string path = FileUtil::getSysTempDirectoryPath() + "tiledump";
```
[the remark this answers is not in the document]
Comment (Minion3665) [reply kit/Delta.hpp]: yep, you're right, good catch!

I've added the slash, thanks :)

**Review by OFFTKP — CHANGES_REQUESTED** [review]

**Review by OFFTKP — APPROVED** [review]
LGTM

**Pull request merged**